### PR TITLE
Add file server endpoint

### PR DESCRIPTION
**Pull request by patrickelectric**

### CODE REVIEW

File: src/drivers/file/server.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+            if (reader.buffer().is_empty()) {
+                info!("End of file reached");
+                break;
+            }
+
+            if (reader.buffer()[0] != mavlink::MAV_STX_V2) {
+                warn!("Invalid MAVLink start byte, skipping");
+                continue;
+            }
```
Comment (joaoantoniocardoso) [inline src/drivers/file/server.rs:52]: we can ditch this: the read_v2_raw_message_async already handles this

File: src/drivers/file/server.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        server: FileServer,
+        mut reader: tokio::io::BufReader<tokio::fs::File>,
+        hub_sender: Arc<broadcast::Sender<Protocol>>,
+    ) -> Result<()> {
+        let source_name = server.path.as_path().display().to_string();
+        loop {
+            let Ok(us_since_epoch) = reader.read_u64().await else {
+                info!("End of file reached");
+                break;
+            };
```
Comment (joaoantoniocardoso) [inline src/drivers/file/server.rs:37]: why do we need this?

File: src/drivers/file/server.rs (hunk 1 above)
Comment (patrickelectric) [reply src/drivers/file/server.rs:52]: Check comment.

File: src/drivers/file/server.rs (hunk 2 above)
Comment (patrickelectric) [reply src/drivers/file/server.rs:37]: Check comment.

**Review by joaoantoniocardoso — APPROVED** [review]
nice!

**Pull request merged**